### PR TITLE
[RHOAIENG-12621] Update the updater github action to the latest branch

### PR DESCRIPTION
**Pull request by jstourac**
This updates the digest updater GHA scripts for both regular workbench images and also runtime images.

On top of that, the runtime images GHA script was a bit enhanced and cleaned of a couple of shellcheck warnings (I can put it as a separate PR if it will be preferred).

https://issues.redhat.com/browse/RHOAIENG-12621

## How Has This Been Tested?
Started relevant GHA via https://github.com/jstourac/notebooks/actions

Standard workbench images:
* https://github.com/jstourac/notebooks/actions/runs/11057100656
* Resulting PR - TBD

Runtime images:
* https://github.com/jstourac/notebooks/actions/runs/11057072140
* Resulting PR - TBD

---

The GHAs failed as we don't have any 2024b builds in https://quay.io/repository/opendatahub/workbench-images?tab=tags&tag=latest --- will re-run once we have some.

Note: one can see results of the openshift-ci jobs here https://prow.ci.openshift.org/?repo=opendatahub-io%2Fnotebooks.

Also, at the moment there is one extra commit in this PR that replaces names images for version N from python-3.9 to python-3.11 because otherwise it would still search for the python-3.9 based images! Version N-1 should be good without this hack. As a result, I plan to incorporate the update for the images as a part of this PR. This hack won't be necessary in the followup updates until we change the python version again.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by jstourac** [comment]
This is waiting for this to be merged openshift/release#57392 - then once it's merged, we need to have another build of the 2024b images (let's re-sync to this branch again).

**Comment by harshad16** [comment]
Thanks for the excellent work.
i will move this from DRAFT to PR.
and use the newly updated changes to update the change.
if anything fails will open a follow PR.

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve

thanks :+1:

**Pull request merged**
